packages/dgrid2/src/ErrorBoundary.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { Component, ErrorInfo, ReactNode } from "react";
+
+interface Props {
+    children: ReactNode;
+    fallback?: ReactNode;
+}
+
+interface State {
+    hasError: boolean;
+    error: Error | null;
+}
+
+class ErrorBoundary extends Component<Props, State> {
+    public state: State = {
+        hasError: false,
+        error: null
+    };
+
+    public static getDerivedStateFromError(error: Error): State {
+        return { hasError: true, error };
+    }
+
+    public componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+        console.error("Error caught by boundary:", error);
+        console.error("Component stack:", errorInfo.componentStack);
+    }
+
+    private handleReset = () => {
+        this.setState({ hasError: false, error: null });
+    };
+
+    public render() {
+        if (this.state.hasError) {
+            if (this.props.fallback) {
+                return this.props.fallback;
+            }
+
+            return (
+                <div className="min-h-[200px] flex items-center justify-center p-8 bg-red-50 rounded-lg border border-red-100">
+                    <div className="text-center">
+                        <h2 className="text-xl font-semibold text-red-700 mb-2">Something went wrong</h2>
+                        <p className="text-red-600">{this.state.error?.message}</p>
+                        <button
+                            onClick={this.handleReset}
+                            className="mt-4 px-4 py-2 bg-red-600 text-white rounded-md hover:bg-red-700 transition-colors"
+                        >
+                            Try again
+                        </button>
+                    </div>
+                </div>
+            );
+        }
+
+        return this.props.children;
+    }
+}
+
+export default ErrorBoundary;
```

Comment:
More of just a style thing, but I don't recall seeing this in any other packages